### ISSUE TITLE
about COSINE endmodel

### ISSUE THREAD

**Issue by Gnaiqing** [JieyuZ2/wrench#33]
Hi @JieyuZ2 and @yinxiangshi , I am trying to run the COSINE endmodel and have some questions about it. The first thing is about the loss function, the `calc_loss` function implemented in wrench computes sum over whole batch, but the original paper computes average, which I think may be an implementation mistake in wrench. Another question is that I am having troubles in reproducing the results in COSINE paper. Although I tried to use the suggested hyperparameters I still get marginal benefit from wrench, and I'm not sure where is wrong. Can you share the scripts you used when evaluating COSINE? Thanks.
